synapse/rest/media/v1/config_resource.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Will Hunt <will@half-shot.uk>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from twisted.web.server import NOT_DONE_YET
+from twisted.web.resource import Resource
+from synapse.http.server import respond_with_json, respond_with_json_bytes, wrap_json_request_handler
+
+
+class MediaConfigResource(Resource):
+    isLeaf = True
+
+    def __init__(self, hs):
+        Resource.__init__(self)
+        config = hs.get_config()
+        self.limits_dict = {
+            "m.upload.size": config.max_upload_size,
+        }
+
+    @wrap_json_request_handler
```

Comment:
hum; does this actually work?

the docstring on `wrap_json_request_handler` says:

>    The handler method must have a signature of "handle_foo(self, request)",
    where "self" must have a "clock" attribute (and "request" must be a
    SynapseRequest).

I don't think the first is true. You might need a `self.clock = hs.get_clock()` in the constructor.